Sources/Authentication/AppAuthSession.swift

```diff
@@ -70,16 +72,27 @@ public final class AppAuthSession: LoginSession {
         }
     }
     
-    
+    /// Finishes the login dialog
+    ///
+    /// - Parameters:
+    ///     - callback: the URL to an authorization endpoint you started the flow with.
+    ///     - endpoint: the token endpoint to query to receive tokens back as the final action in the flow.
     @MainActor
-    public func finalise(callback url: URL) async throws -> TokenResponse {
+    public func finalise(callback url: URL, endpoint: URL) async throws -> TokenResponse {
         flow?.resumeExternalUserAgentFlow(with: url)
         
-        guard let authorizationCode else {
-            throw LoginError.inconsistentStateResponse
+        if let error {
+            if error.localizedDescription.contains("State mismatch") {
+                throw LoginError.inconsistentStateResponse
+            }
+            throw error
         }
+        
+        guard let authorizationCode else { throw LoginError.missingAuthorizationCode }
+        guard let redirectURI else { throw LoginError.missingRedirectURI }
+        
         return try await service
-            .fetchTokens(authorizationCode: authorizationCode)
+            .fetchTokens(authorizationCode: authorizationCode, redirectURI: redirectURI, endpoint: endpoint)
```

Comment:
My spidey senses are going a little bit here... Why are we using our own network package to do the token exchange...? Shouldn't we use AppAuth's functionality to do this? I don't completely get the syntax yet but this section of the AppAuth documentation seems pertinent: https://github.com/openid/AppAuth-iOS#making-api-calls

Reply:
If there's app auth functionality to do this, by all means let's use it 💯